connection.go

```diff
@@ -62,15 +64,55 @@ func NewConnection(addr, user, pass string) *Connection {
 	}
 }
 
+func NewTLSConnection(addr, user, pass string, certPool *x509.CertPool) *Connection {
+	connection := NewConnection(addr, user, pass)
+	connection.dial = func(network, address string) (net.Conn, error) {
+		conn, err := net.DialTimeout(network, address, 5*time.Second)
+		if err != nil {
+			return nil, err
+		}
+
+		// TODO Save Info packet somewhere accessible
+		br := bufio.NewReaderSize(conn, 32768)
+		_, err = Parse(br)
+		if err != nil {
+			return conn, err
```

Comment:
probably shouldn't return `conn` (here and 84), just to avoid somebody accidentally ignoring an error and then blindly using an unexpectedly insecure connection